src/transformers/models/deformable_detr/modeling_deformable_detr.py

```diff
@@ -664,19 +598,8 @@ def forward(
             )
         else:
             raise ValueError(f"Last dim of reference_points must be 2 or 4, but got {reference_points.shape[-1]}")
-        try:
-            # GPU
-            output = MultiScaleDeformableAttentionFunction.apply(
-                value,
-                spatial_shapes,
-                level_start_index,
-                sampling_locations,
-                attention_weights,
-                self.im2col_step,
-            )
-        except Exception:
-            # CPU
-            output = ms_deform_attn_core_pytorch(value, spatial_shapes, sampling_locations, attention_weights)
+
+        output = ms_deform_attn_core_pytorch(value, spatial_shapes, sampling_locations, attention_weights)
```

Comment:
So we're just switching to the CPU implementation here? 

The Deformable DETR authors recommended to only use this for debugging: 

> The cuda version is much faster and more memory efficient than this pytorch version.

See thread: https://github.com/fundamentalvision/Deformable-DETR/issues/9

Reply:
Oh I initially thought it was a new function that was vectorized, not the CPU implementation. Let's not remove it indeed.

Reply:
In that case, can't I use this CUDA version for Mask2Former as well? In fact even OneFormer could benefit from it if it's okay to use this CUDA implementation.

Reply:
cc'ing @praeclarumjj3 (OneFormer author) here. 

It would probably be best to provide the custom CUDA kernels as an option, rather than by default. This way, we can run the model in Google Colab, which is not the case at the moment.

We did the same for YOSO. This model has a boolean attribute in the config to determine whether or not to use the custom kernel.

Reply:
@NielsRogge Agreed, it makes sense to keep it as optional. I can take up this change today for Mask2Former if you and @alaradirik think it makes sense to go ahead and add support for the CUDA implementation too. Let me know.

Reply:
Hi @NielsRogge, I included the CUDA version in the initial push for OneFormer. However, @alaradirik suggested we remove that. Do you want me to push it again? 

>This model has a boolean attribute in the config to determine whether or not to use the custom kernel.

That's an excellent idea. We can do that.

Reply:
It is not the cpu implementation, it can still be run on GPU. The PR simply removes the custom CUDA kernels, which parallelizes some of the GPU operations but also prevent users with a incompatible cuda library version to run the model on GPU.

Reply:
With that said, adding a boolean attribute to use the custom kernel is a good idea but users would need to first download the configuration of the checkpoint, edit it and then initialize the model if it's not going to be the default behaviour.

@NielsRogge @sgugger

Reply:
Hmm normally if you just do:
```
from transformers import DeformableDetrForObjectDetection

model = DeformableDetrForObjectDetection.from_pretrained("SenseTime/deformable-detr", use_custom_kernel=False) 
```

then it should work. By providing additional kwargs to the `from_pretrained` method, you can edit the configuration.

Reply:
And @shivalikasingh95 sure but let's add support for the custom kernel in a separate PR for Mask2Former and OneFormer. We can set it to False by default, and add a boolean attribute for those models.

For Deformable DETR, we could change the behaviour to set it to False by default and inform users that if they really want to use the kernel, set the boolean attribute to True.

Reply:
@alaradirik The function will be a lot slower on GPU since it's not parallelized, so enabling the custom CUDA kernel when possible would be better.

I don't think a flag is needed in the config. Let's just not fail if trying to load the CUDA kernel doesn't work and default to this non-vectorized implementation.